numpy/core/src/multiarray/abstractdtypes.c

```diff
@@ -13,6 +13,12 @@
 #include "common.h"
 
 
+static NPY_INLINE PyArray_Descr *
+int_default_descriptor(PyArray_DTypeMeta* NPY_UNUSED(cls))
+{
+    return PyArray_DescrFromType(NPY_LONG);
+}
+
```

Comment:
All DTypes should have a `default_descr()` class method (even these abstract ones). I guess it could make sense to not have one in some cases, but here it is required.